### PR TITLE
Fix builds

### PR DESCRIPTION
**Pull request by Xerkus**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description
Fixing tests broken during Laminas migration

### CODE REVIEW

File: src/Headers.php [hunk 1]
```diff
@@ -283,7 +283,9 @@ public function get($name)
             return false;
         }
 
-        $class = ($this->getPluginClassLoader()->load(str_replace('-', '', $key))) ?: 'Laminas\Http\Header\GenericHeader';
+        $class = (
+            $this->getPluginClassLoader()->load(str_replace('-', '', $key))
+        ) ?: 'Laminas\Http\Header\GenericHeader';
```
Comment (michalbundyra) [inline src/Headers.php]: Brackets around expression are redundant, so I'd suggest the following formatting:
```php
 $class = $this->getPluginClassLoader()->load(str_replace('-', '', $key))
    ?: 'Laminas\Http\Header\GenericHeader';
```

Comment (Xerkus) [reply src/Headers.php]: They are. They increase readability however, in my opinion

**Comment by Xerkus** [comment]
All tests failed successfully.

Remaining failures are linked to ssl disabled by default in test web server.
```
2) LaminasTest\Http\Client\ProxyAdapterTest::testUsesProvidedArgSeparator
stream_socket_enable_crypto(): SSL operation failed with code 1. OpenSSL Error messages:
error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure

/home/travis/build/laminas/laminas-http/src/Client/Adapter/Proxy.php:289
/home/travis/build/laminas/laminas-http/src/Client/Adapter/Proxy.php:165
/home/travis/build/laminas/laminas-http/src/Client.php:1457
/home/travis/build/laminas/laminas-http/src/Client.php:945
/home/travis/build/laminas/laminas-http/test/Client/CommonHttpTests.php:1090
/home/travis/build/laminas/laminas-http/test/Client/ProxyAdapterTest.php:125
```
 I believe proper way to fix it would be to remove ssl and use tls only. 
https://github.com/laminas/laminas-http/blob/2f5982ea8d72c9660c78145cf7e4edb9d0fc0e01/src/Client/Adapter/Proxy.php#L280-L285

------

Following curl failure needs to be debugged further, potential bug exposure:
```
1) LaminasTest\Http\Client\CurlTest::testNoCaseSensitiveHeaderName

Laminas\Http\Client\Adapter\Exception\RuntimeException: Error in cURL request: transfer closed with outstanding read data remaining
/home/travis/build/laminas/laminas-http/src/Client/Adapter/Curl.php:475
/home/travis/build/laminas/laminas-http/src/Client.php:1457
/home/travis/build/laminas/laminas-http/src/Client.php:945
/home/travis/build/laminas/laminas-http/test/Client/CurlTest.php:453
```

**Comment by michalbundyra** [comment]
@Xerkus  I am really unsure about failing tests here.
See the latest results of zend-http:
https://travis-ci.org/zendframework/zend-http

All was green there (18h ago last run). I think it might be something wrong with the migration here.

**Comment by Xerkus** [comment]
Difference is test uses `$request->setUri('https://getlaminas.org');`
And  previously it was a server for zf site which supported ssl v3.

**Comment by Xerkus** [comment]
It would have been easier if I bothered to setup test webserver locally in docker.

Newer curl complains about chunked  transfer that was not properly terminated with zero length chunk. Fixed test asset to emit correct chunked  response.

**Pull request closed without merging**